doc/RELEASE_WALKTHROUGH.rst.txt

```diff
@@ -25,95 +46,151 @@ check it out::
-    $ ./wheel-uploader -r warehouse -u $CDN_URL -s -v -w ~/wheelhouse -t manylinux1 numpy 1.12.0
-    $ ./wheel-uploader -r warehouse -u $CDN_URL -s -v -w ~/wheelhouse -t macosx numpy 1.12.0
+    $ NPY_WHLS=../numpy/release/installers
+    $ ./wheel-uploader -u $CDN_URL -n -v -w $NPY_WHLS -t win numpy 1.14.5
+    $ ./wheel-uploader -u $CDN_URL -n -v -w $NPY_WHLS -t manylinux1 numpy 1.14.5
+    $ ./wheel-uploader -u $CDN_URL -n -v -w $NPY_WHLS -t macosx numpy 1.14.5
 
-If one of the commands breaks in the middle, you may need to manually upload
-the rest of the files using twine because wheel-uploader always starts from the
-beginning and PyPI does not allow the same file to be uploaded twice.
+If you do this often, consider making CDN_URL and NPY_WHLS part of your default
+environment. Note that we need local copies of the files in order to generate
+hashes to include in the README files generated later.
 
-The source files should be uploaded last to avoid problems. PyPI will only allow
-a single source distribution, here it is the zip archive::
+Tag the release
+---------------
 
-    $ cd ../numpy
-    $ twine upload -s release/installers/numpy-1.12.0.zip
+Once the wheels have been built and downloaded without errors, go back to your
+numpy repository in the maintenance branch and tag the ``REL`` commit, signing
+it with your gpg key, and build the source distribution archives::
 
-If this is not a final release, log into PyPI and hide the new directory while
-making sure the last final release is visible.
+    $ git tag -s v1.14.5
 
-Upload tarballs to github
--------------------------
-We want to post the hash values for the files uploaded to PyPI in the release
-generated README file, so first copy the wheels into the installers directory
-and remove the now unneeded signature files::
+You should upload your public gpg key to github, so that the tag will appear
+"verified" there.
 
-    $ cp ~/wheelhouse/numpy-1.12.0* release/installers
-    $ rm release/installers/*.asc
+Check that the files in ``release/installers`` have the correct versions, then
+push the tag upstream::
+
+    $ git push upstream v1.14.5
+
+We wait until this point to push the tag because it is very difficult to change
+the tag after it has been pushed.
+
+Reset the maintenance branch into a development state
+-----------------------------------------------------
+
+Add another ``REL`` commit to the numpy maintenance branch, which resets the
+``ISREALEASED`` flag to ``False`` and increments the version counter::
 
-Generate the ``release/Changelog`` and ``release/README`` files::
+    $ gvim pavement.py setup.py
+    $ git commit -a -m"REL: prepare 1.14.x for further development"
+    $ git push upstream maintenance/1.14.x
+
+This strategy is copied from the scipy release procedure and was used in numpy
+for the first time in 1.14.3. It needed to be modified a little since numpy
+has more strict requirements for the version number.
```

Comment:
not sure what these more strict requirements are, I think all policies are identical?

Reply:
No, numpy has a test [here](https://github.com/numpy/numpy/blob/master/numpy/tests/test_numpy_version.py) which limits what can go in the version number. See how I tried to make the version number "1.14-dev" in the second to last commit [here](https://github.com/numpy/numpy/commits/b8df705bdcce92d3e2c6f050eb4414192cf0df04), like I saw you do in scipy, but numpy didn't like it, so I had to push a followup commit.

That's why I wrote this here.

Reply:
Ah that. That's the same for SciPy, there's just no test for it. `1.14-dev` is not a valid version (says PEP440), so I don't think SciPy has had such a thing recently.

I think the sentence just raises unnecessary questions, it's better to remove it. If you want to keep something here, I suggest a link to https://docs.scipy.org/doc/scipy/reference/dev/index.html#version-numbering or to copy that info over to the NumPy devguide.

Reply:
And I should check before I write next time: scipy has the same test, https://github.com/scipy/scipy/blob/master/scipy/_build_utils/tests/test_scipy_version.py

Reply:
OK, that makes sense. I could have sworn I saw a REL commit in scipy that manually added `-dev`, but maybe I imagined it.